aiohttp/http_writer.py

```diff
@@ -56,13 +59,18 @@ def _write(self, chunk):
             raise asyncio.CancelledError('Cannot write to closing transport')
         self._transport.write(chunk)
 
-    def write(self, chunk, *, drain=True, LIMIT=64*1024):
+    def write(self, chunk, *, drain=True, LIMIT=64 * 1024):
         """Writes chunk of data to a stream.
 
         write_eof() indicates end of stream.
         writer can't be used after write_eof() method being called.
         write() return drain future.
         """
+        self.on_chunk_sent.freeze()
```

Comment:
I pretty sure there is better place for signal freezing than first write attempt.

Reply:
Actually I was quite surprised with the whole concept of `freeze()`. It's very tricky! Do you know why it's necessary at all ? 
The only reason I can think of is that the author wanted to prevent modifications to list of handlers while processing `send()`. 

Still, it's possible that we will decide to remove this `Signal` altogether, but if we don't, what do you think is the correct place to call freeze() ? 
- I can't do this in `__init__` because I want to add handlers later
- I can't delegate this responsibility to the user of the class, because this introduces a braking change where a code which works now like:
  ```
  writer = StreamWriter(...)
  await writer.write(...)
  ```
  will start raising `RuntimeError`